doc/lsst.daf.butler/datastores.rst

```diff
@@ -0,0 +1,79 @@
+.. _daf_butler-datastores:
+
+#######################
+Datastore Configuration
+#######################
+
+A Butler `~lsst.daf.butler.Datastore` is configured in the ``datastore`` section of the top-level Butler YAML configuration.
+The only mandatory entry in the datastore configuration is the ``cls`` key.
+This specifies the fully qualified class name of the Python class implementing the datastore.
+The default is to use the `~lsst.daf.butler.datastores.posixDatastore.PosixDatastore`.
+All other keys depend on the specific datastore class that is selected.
+
```

Comment:
It would be nice up here at the top in the intro to put a list of the supported datastores (either class names or brief text description) and a link to the details below.